### PR TITLE
Seperate flame and topRegion rendering, for better code readability.

### PR DESCRIPTION
**Pull request by TechFish3000**
if you would like me to stop suggesting this, just say so.
The only reason i want this is so i can have a `topRegion` but not a flame.

### CODE REVIEW

**Comment by DeltaNedas** [comment]
set flame color to be empty
it already exists

**Comment by TechFish3000** [comment]
yes, but that gets rid of the `topRegion`

**Comment by TechFish3000** [comment]
Would i use `Null` or `None` to check if `topRegion` exists?

File: core/src/mindustry/world/blocks/production/GenericSmelter.java [hunk 1]
```diff
@@ -41,7 +49,7 @@ public void draw(Tile tile){
             Draw.tint(flameColor);
             Fill.circle(tile.drawx(), tile.drawy(), 3f + Mathf.absin(Time.time(), 5f, 2f) + cr);
             Draw.color(1f, 1f, 1f, entity.warmup);
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/GenericSmelter.java]: this line is now duplicate, you might want to move it above the first if, and call reset when both are done 👀

File: core/src/mindustry/world/blocks/production/GenericSmelter.java [hunk 2]
```diff
@@ -30,7 +30,15 @@ public void draw(Tile tile){
...
+        //draw topRegion
+        if(entity.warmup > 0f){
+            Draw.color(1f, 1f, 1f, entity.warmup);
+            Draw.rect(topRegion, tile.drawx(), tile.drawy());
+            Draw.color();
+        };
+        
+        
+        //draw "flame" in center
         if(entity.warmup > 0f && flameColor.a > 0.001f){
```
Comment (PetrGasparik) [inline core/src/mindustry/world/blocks/production/GenericSmelter.java]: you may enclose this `if` into the upper one and get rid of "entity.warmup > 0f &&" here, plus one less check

**Comment by TechFish3000** [comment]
There, it is efficient now (hopefully).

**Comment by PetrGasparik** [comment]
> There, it is efficient now (hopefully).

I approve! :)

**Comment by Anuken** [comment]
This has some incorrect indentation and isn't compatible with 6.0's entity rewrite.

**Pull request closed without merging**